sympy/physics/mechanics/kane.py

```diff
@@ -923,36 +925,57 @@ def kanes_equations(self, bodies, loads=None):
 
         return (self._fr, self._frstar)
 
+    def _kanes_equations_not_called(self, attr):
+        if not self._fr or not self._frstar:
+            msg = 'kanes_equations() must be run before {} is available.'
+            raise AttributeError(msg.format(attr))
+
     def rhs(self, inv_method=None):
-        """Returns the system's equations of motion in first order form.
+        """Returns the right hand side of the explicit first order full
+        differential equations of motion where:
 
-        The output of this will be the right hand side of:
+            x' = M^-1 * f
 
-        [qdot, udot].T = f(q, u, t)
+        where x, the state vector, contains the independent and dependent
+        generalized coordinates and speeds::
 
-        Or, the equations of motion in first order form.  The right hand side
-        is what is needed by most numerical ODE integrators.
+            x = | q_s |
+                | q_r |
+                | u_s |
+                | u_r |
+
+        The right hand side of these equations is what is needed by most
+        numerical ODE/DAE integrators.
 
         Parameters
         ==========
 
-        inv_method : str
-            The specific sympy inverse matrix calculation method to use. For a
-            list of valid methods, see
-            :meth:`~sympy.matrices.matrices.MatrixBase.inv`
+        inv_method : str, {'CH', 'GE', 'LU', or 'ADJ'}, optional
+            If None LU decomposition is used to solve for x'. If `CH`, Cholesky
+            decomposition is used. Cholesky decomposition can almost always be
+            used, as long as the mass matrix is semi-positive definite. The
+            other methods are not recommended as they operate stricyly on the
+            mass matrix and attempt to compute the inverse. In general, these
+            are not computationally efficient and should only be used for very
+            small problems. The later three come from
+            :meth:`~sympy.matrices.matrices.MatrixBase.inv`.
 
         """
+        qdot = Matrix([self._qdot_u_map[qd] for qd in self.q])
         if inv_method is None:
-            self._rhs = self.mass_matrix_full.LUsolve(self.forcing_full)
+            udot = self.mass_matrix.LUsolve(self.forcing)
+        elif inv_method == 'CH':
+            udot = self.mass_matrix.cholesky_solve(self.forcing)
         else:
-            self._rhs = (self.mass_matrix_full.inv(inv_method,
-                         try_block_diag=True) * self.forcing_full)
+            udot = (self.mass_matrix.inv(method=inv_method,
+                                         try_block_diag=True) * self.forcing)
```

Comment:
Ooh neat. I know you've mentioned that this method can be slow in the past, so getting rid if a lot of the calculations is great!